libibverbs/verbs.c

```diff
@@ -701,7 +701,22 @@ int ibv_query_qp_data_in_order(struct ibv_qp *qp, enum ibv_wr_opcode op,
 	 */
 	return 0;
 #else
-	return get_ops(qp->context)->query_qp_data_in_order(qp, op, flags);
+	uint32_t query_flags;
+	int result;
+
+	query_flags = flags ? flags : IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS;
+
+	result = get_ops(qp->context)->query_qp_data_in_order(qp, op, query_flags);
+	if (query_flags == IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS) {
+		if (result & IBV_QUERY_QP_DATA_IN_ORDER_WHOLE_MSG ||
+		    get_ops(qp->context)->query_qp_data_in_order(qp, op, 0))
```

Comment:
There is only one other implementation just fix it to return the new style always

Reply:
Ok, done.